internal/core/docker_context.go

```diff
@@ -0,0 +1,328 @@
+		if fi.IsDir() {
+			if isContextDir(filepath.Join(root, fi.Name())) {
+				result = append(result, fi.Name())
+			}
+
+			subs, err := listRecursivelyMetadataDirs(filepath.Join(root, fi.Name()))
+			if err != nil {
+				return nil, err
+			}
+
+			for _, s := range subs {
+				result = append(result, filepath.Join(fi.Name(), s))
+			}
+		}
+	}
+
+	return result, nil
+}
+
+func parseTypedOrMap(payload []byte, getter typeGetter) (any, error) {
+	if len(payload) == 0 || string(payload) == "null" {
+		return nil, nil
+	}
+
+	if getter == nil {
+		var res map[string]any
+		if err := json.Unmarshal(payload, &res); err != nil {
+			return nil, err
+		}
+		return res, nil
+	}
+
+	typed := getter()
+	if err := json.Unmarshal(payload, typed); err != nil {
+		return nil, err
+	}
+
+	return reflect.ValueOf(typed).Elem().Interface(), nil
+}
+
+// getHomeDir returns the home directory of the current user with the help of
+// environment variables depending on the target operating system.
+// Returned path should be used with "path/filepath" to form new paths.
+//
+// On non-Windows platforms, it falls back to nss lookups, if the home
+// directory cannot be obtained from environment-variables.
+//
+// If linking statically with cgo enabled against glibc, ensure the
+// osusergo build tag is used.
+//
+// If needing to do nss lookups, do not disable cgo or set osusergo.
+//
+// getHomeDir is a copy of [pkg/homedir.Get] to prevent adding docker/docker
+// as dependency for consumers that only need to read the config-file.
+//
+// [pkg/homedir.Get]: https://pkg.go.dev/github.com/docker/docker@v26.1.4+incompatible/pkg/homedir#Get
+func getHomeDir() string {
+	home, _ := os.UserHomeDir()
+	if home == "" && runtime.GOOS != "windows" {
+		if u, err := user.Current(); err == nil {
+			return u.HomeDir
+		}
+	}
+	return home
+}
+
+// configurationDir returns the directory the configuration file is stored in
+func configurationDir() string {
+	configDir := os.Getenv(envOverrideConfigDir)
+	if configDir == "" {
+		configDir = filepath.Join(getHomeDir(), configFileDir)
+	}
+	return configDir
+}
+
+// GetDockerHostFromCurrentContext returns the Docker host from the current Docker context.
+// For that, it traverses the directory structure of the Docker configuration directory,
+// looking for the current context and its Docker endpoint.
+func GetDockerHostFromCurrentContext() (string, error) {
+	metaRoot := filepath.Join(filepath.Join(configurationDir(), contextsDir), metadataDir)
+
+	ms := &metadataStore{
+		root:   metaRoot,
+		config: newConfig(func() any { return &dockerContext{} }, defaultStoreEndpoints...),
+	}
+
+	md, err := ms.list()
+	if err != nil {
+		return "", err
+	}
+
+	currentContext := currentContext()
+
+	for _, m := range md {
+		if m.Name == currentContext {
+			ep, ok := m.Endpoints[dockerEndpoint].(endpointMeta)
+			if ok {
+				return ep.Host, nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("no Docker host found in the current context")
```

Comment:
suggestion: avoid the overhead of `fmt.Errorf` when not needed.
```suggestion
	return "", errors.New("no Docker host found in the current context")
```